### PR TITLE
feat(template): Fix support for SPOT instances when preemptible is enabled

### PR DESCRIPTION
**Pull request by gcaracuel**
Provider version `4.11.0` added field `provisioning_model` to `scheduling` section. 
This field provides access to https://cloud.google.com/spot-vms

Unfortunately API does not support value to be set to `STANDARD` when `preemptible` is true

```
Error: Error creating instance template: googleapi: Error 400: Invalid value for field 'resource.properties.scheduling.provisioningModel': 'STANDARD'. For preemptible, only allowed provisioning_model value is SPOT., invalid
```

SPOT instances are a good way to save some costs when applications supports this which is quite common on autoscaling instances.

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_generate_modules
 The modules need to be regenerated. Please run `make_build`. 
 ```bash 
Checking submodule's files generation
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=.git' /workspace/modules/instance_template/README.md /tmp/tmp.xXvwJnYqYs/workspace/modules/instance_template/README.md
39a40
> | provisioning\_model | Describe the type of preemptible VM | `string` | `"STANDARD"` | no |
Error: submodule's files generation has not been run, please run the
'make build' command and commit changes
 ``` 
- ⚠️check_documentation
 The documentation needs to be regenerated. Please run `make generate_docs`. 
 ```bash 
Checking for documentation generation
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=.git' '--exclude=autogen' '--exclude=*.tfvars' /workspace/modules/instance_template/README.md /tmp/tmp.nc1BRqSPGU/generate_docs/workspace/modules/instance_template/README.md
39a40
> | provisioning\_model | Describe the type of preemptible VM | `string` | `"STANDARD"` | no |
Error: Documentation generation has not been run, please run the
'make docker_generate_docs' command and commit the above changes.
 ```

File: modules/instance_template/variables.tf [hunk 1]
```diff
@@ -71,6 +71,16 @@ variable "on_host_maintenance" {
   default     = "MIGRATE"
 }
 
+variable "provisioning_model" {
```
Comment (bharathkkb) [inline modules/instance_template/variables.tf:74]: This variable does not seem to be used since we switch local.provisioning_model?

**Review by bharathkkb — COMMENTED** [review]
Thanks for the PR @gcaracuel

File: modules/instance_template/variables.tf (hunk 1 above)
Comment (gcaracuel) [reply modules/instance_template/variables.tf]: My bad, that was a leftover from a previous implementation, when preemtible is used only 1 valid value for this parameter so moved to a local.

I have edited the title on this PR to fix actual change

File: modules/instance_template/main.tf [hunk 2]
```diff
@@ -146,9 +150,11 @@ resource "google_compute_instance_template" "tpl" {
...
 
   scheduling {
-    preemptible         = var.preemptible
-    automatic_restart   = local.automatic_restart
-    on_host_maintenance = local.on_host_maintenance
+    preemptible                 = var.preemptible
+    automatic_restart           = local.automatic_restart
+    on_host_maintenance         = local.on_host_maintenance
+    provisioning_model          = local.provisioning_model
+    instance_termination_action = "STOP"
```
Comment (bharathkkb) [inline modules/instance_template/main.tf:157]: One of the tests seem to be failing. It seems like we can only set instance_termination_action STOP if using spot. 
```
       
       Error: Error creating instance template: googleapi: Error 400: Invalid value for field 'resource.properties.scheduling.instanceTerminationAction': 'STOP'. You cannot specify a termination action for a VM instance that has the standard provisioning model (default). To use instance termination action, the VM instance must use the Spot provisioning model., invalid
       
         with module.instance_template_simple.module.instance_template.google_compute_instance_template.tpl,
         on ../../../../modules/instance_template/main.tf line 66, in resource "google_compute_instance_template" "tpl":
         66: resource "google_compute_instance_template" "tpl" {
         ```

**Pull request closed without merging**